libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/msd/validator/StaticWorkloadValidatorFactory.java

```diff
@@ -25,14 +28,14 @@
     /**
      * Creates the static workload validators
      */
-    void create();
+    void create(StaticWorkloadDataRepositoryFactory repositoryFactory);
```

Comment:
this looks wrong from design perspective. Why are we passing a factory into another factory. Is it just a wrong choice of class name where the StaticWorkloadDataRepositoryFactory is not a factory but some other class?

Reply:
StaticWorkloadValidatorFactory implementation will initiate different types of Validators and each validator will receive its own Repository e.g. VIPValidator would take VIPRepository as initialization parameter. Hence StaticWorkloadFactory is receiving another Factory as a parameter.